### PR TITLE
Bug 1710828 - Document InvalidType error [doc only]

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: docs/user/reference/metrics/counter.md [hunk 1]
```diff
@@ -100,7 +100,8 @@ Increases the counter by a certain amount. If no amount is passed it defaults to
 
 #### Recorded errors
 
-* [`invalid_value`](../../user/metrics/error-reporting.md): If the counter is incremented by `0` or a negative value.
+* [`invalid_value`](../../user/metrics/error-reporting.md): if the counter is incremented by `0` or a negative value.
```
[the remark this answers is not in the document]
Comment (badboy) [reply docs/user/reference/metrics/counter.md:103]: or at least have it consistent across everything.

File: docs/user/reference/metrics/quantity.md [hunk 2]
```diff
@@ -96,7 +96,8 @@ Glean.display.width.set(innerHeight);
 
 #### Recorded errors
 
-* [`invalid_value`](../../user/metrics/error-reporting.md): If a negative value is passed in.
+* [`invalid_value`](../../user/metrics/error-reporting.md): if a negative value is passed in.
+* [`invalid_type`](../../user/metrics/error-reporting.md): if a negative, floating point or non-number value is given.
```
Comment (badboy) [inline docs/user/reference/metrics/quantity.md:100]: as above: negative should result in `invalid_value`.

File: docs/user/reference/metrics/string_list.md [hunk 3]
```diff
@@ -182,6 +183,7 @@ Glean.search.engines.set(["wikipedia", "duck duck go"]);
 
 * `invalid_overflow`: if the string is too long. (Prior to Glean 31.5.0, this recorded an `invalid_value`).
 * `invalid_value`: if the list is too long.
+* [`invalid_type`](../../user/metrics/error-reporting.md): if a non-string array value is given.
```
Comment (badboy) [inline docs/user/reference/metrics/string_list.md:186]: Should the other `invalid_*` above be linked too?

Comment (badboy) [reply docs/user/reference/metrics/string_list.md:186]: I'd leave out the value.

File: docs/user/reference/metrics/timespan.md [hunk 4]
```diff
@@ -472,10 +472,9 @@ function afterLogin(aDuration) {
 
 #### Recorded errors
 
-* [`invalid_value`](../../user/metrics/error-reporting.md):
-  If you attempt to record a negative elapsed duration.
-* [`invalid_state`](../../user/metrics/error-reporting.md):
-  If you call this API after calling `start` or call this API multiple times.
+* [`invalid_value`](../../user/metrics/error-reporting.md): if you attempt to record a negative elapsed duration.
+* [`invalid_state`](../../user/metrics/error-reporting.md): if you call this API after calling `start` or call this API multiple times.
+* [`invalid_type`](../../user/metrics/error-reporting.md): if a negative, floating point or non-number value is given.
```
Comment (badboy) [inline docs/user/reference/metrics/timespan.md:477]: as above. negative is `invalid_value`.

File: docs/user/reference/metrics/uuid.md [hunk 5]
```diff
@@ -193,6 +193,7 @@ Glean.user.clientId.set(uuid);
 #### Recorded errors
 
 * [`invalid_value`](../../user/metrics/error-reporting.md): if the value is set to a string that is not a UUID (only applies for dynamically-typed languages, such as Python).
+* [`invalid_type`](../../user/metrics/error-reporting.md): if a non-string value is given.
```
Comment (badboy) [inline docs/user/reference/metrics/uuid.md:196]: Some SDKs expect a proper `Uuid` object. Maybe we can make this more clear?
`if a non-string or non-uuid value is given`

File: docs/user/user/metrics/error-reporting.md [hunk 6]
```diff
@@ -11,6 +11,7 @@ The following categories of errors are recorded:
 - `invalid_label`: The label on a labeled metric was invalid.
 - `invalid_state`: The metric caught an invalid state while recording.
 - `invalid_overflow`: The metric value to be recorded overflows the metric-specific upper range.
+- `invalid_type`: The metric value is not of the expected type. This error type is only recorded by the Glean JavaScript SDK. This error may only happen in dynamically typed languages.
```
Comment (badboy) [inline docs/user/user/metrics/error-reporting.md:14]: ah, here we have the explanation.

File: docs/user/reference/metrics/counter.md [hunk 7]
```diff
@@ -100,7 +100,8 @@ Increases the counter by a certain amount. If no amount is passed it defaults to
...
+* [`invalid_type`](../../user/metrics/error-reporting.md): If a negative, floating point or non-number value is given.
```
[the remark this answers is not in the document]
Comment (brizental) [reply docs/user/reference/metrics/counter.md:104]: Yeap, it was a mistake yo put negative here.

File: docs/user/reference/metrics/timespan.md (hunk 4 above)
Comment (brizental) [reply docs/user/reference/metrics/timespan.md:477]:
```suggestion
* [`invalid_type`](../../user/metrics/error-reporting.md): if a floating point or non-number value is given.
```

File: docs/user/user/metrics/error-reporting.md (hunk 6 above)
Comment (brizental) [reply docs/user/user/metrics/error-reporting.md:14]: Yeap, thought I'd add it here so I don't need to repeat myself in all the metrics files (we would forget to do that everytime, probably). With regards to filing a bug to do this in Python and JS, that is up to you and @chutten I guess. Not sure wha are the priorities for the FOG and Python SDKs right now.

File: docs/user/reference/metrics/uuid.md [hunk 8]
```diff
@@ -193,6 +193,7 @@ Glean.user.clientId.set(uuid);
 #### Recorded errors
 
 * [`invalid_value`](../../user/metrics/error-reporting.md): if the value is set to a string that is not a UUID (only applies for dynamically-typed languages, such as Python).
+* [`invalid_type`](../../user/metrics/error-reporting.md): if a non-string or non-uuid value is given.
```
[the remark this answers is not in the document]
Comment (badboy) [reply docs/user/reference/metrics/uuid.md:196]: (should fix the spellcheck CI failure)

File: docs/user/reference/metrics/string_list.md [hunk 9]
```diff
@@ -95,8 +95,9 @@ Glean.search.engines.add("duck duck go");
 
 #### Recorded errors
 
-* `invalid_overflow`: if the string is too long. (Prior to Glean 31.5.0, this recorded an `invalid_value`).
-* `invalid_value`: if the list is too long.
+* [`invalid_overflow`](../../user/metrics/error-reporting.md): if the string is too long. (Prior to Glean 31.5.0, this recorded an `invalid_value`).
+* [`invalid_value`](../../user/metrics/error-reporting.md): if the list is too long.
+* [`invalid_type`](../../user/metrics/error-reporting.md): If a non-string value is given.
```
Comment (badboy) [inline docs/user/reference/metrics/string_list.md:100]:
```suggestion
* [`invalid_type`](../../user/metrics/error-reporting.md): if a non-string value is given.
```

File: docs/user/reference/metrics/timespan.md [hunk 10]
```diff
@@ -472,10 +472,9 @@ function afterLogin(aDuration) {
 
 #### Recorded errors
 
-* [`invalid_value`](../../user/metrics/error-reporting.md):
-  If you attempt to record a negative elapsed duration.
-* [`invalid_state`](../../user/metrics/error-reporting.md):
-  If you call this API after calling `start` or call this API multiple times.
+* [`invalid_value`](../../user/metrics/error-reporting.md): if you attempt to record a negative elapsed duration.
+* [`invalid_state`](../../user/metrics/error-reporting.md): if you call this API after calling `start` or call this API multiple times.
```
Comment (badboy) [inline docs/user/reference/metrics/timespan.md:476]: we don't use `you` anywhere else, so let's fix it here.

```suggestion
* [`invalid_value`](../../user/metrics/error-reporting.md): if attempting to record a negative elapsed duration.
* [`invalid_state`](../../user/metrics/error-reporting.md): if this method is called after calling `start` or this method is called multiple times.
```

**Review by badboy — APPROVED** [review]

File: docs/user/user/metrics/error-reporting.md (hunk 6 above)
Comment (chutten) [reply docs/user/user/metrics/error-reporting.md:14]: For it to work in Firefox Desktop's JS we'd need https://bugzilla.mozilla.org/show_bug.cgi?id=1691073

**Pull request merged**